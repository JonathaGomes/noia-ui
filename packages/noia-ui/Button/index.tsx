import React from "react";

type ButtonProps = {
  color: string;
  children: React.ReactNode;
};

const Button = ({ color, children }: ButtonProps) => {
  return <button style={{ backgroundColor: color }}>{children}</button>;
};
