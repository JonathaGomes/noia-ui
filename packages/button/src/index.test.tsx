import React from 'react'

import { render, screen } from '@testing-library/react'

import Button from '..'

describe('<Button />', () => {
  it('should render a button', () => {
    const { container } = render(<Button>hello</Button>)

    expect(screen.getByRole('button', { name: /hello/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a red button', () => {
    render(<Button bg="red">hello</Button>)

    screen.logTestingPlaygroundURL()

    expect(screen.getByRole('button', { name: /hello/i })).toHaveStyle('background-color: red')
  })
})