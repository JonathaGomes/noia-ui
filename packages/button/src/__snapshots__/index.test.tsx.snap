// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button 1`] = `
.c0 {
  background-color: #fff;
  padding: 10px 20px;
  border: none;
  cursor: pointer;
}

.c0:hover {
  background-color: #fff;
}

<button
  class="c0"
>
  hello
</button>
`;
